core/src/test/java/io/grpc/internal/PickFirstLeafLoadBalancerTest.java

```diff
@@ -110,46 +125,53 @@ public void uncaughtException(Thread t, Throwable e) {
   private FakeSubchannel mockSubchannel3;
   @Mock
   private FakeSubchannel mockSubchannel4;
+  @Mock
+  private FakeSubchannel mockSubchannel5;
   @Mock // This LoadBalancer doesn't use any of the arg fields, as verified in tearDown().
   private PickSubchannelArgs mockArgs;
 
+  @BeforeClass
+  public static void setUpClass() {
```

Comment:
I didn't see you reset the property @AfterClass, so other tests in the entire grpc test will always see happyeyeballs enabled. However I think we expect them to test on happy eyeballs not enabled case.
More importantly, the test suite here should verify both enabled/disabled cases (use Parameterized), or at least have the original tests do the disabled cases, and have your new tests enable the variable. I didn't realize last time I saw.

Reply:
Changed to setting in `setup` and added an `@after` method to restore it.